### PR TITLE
Fix missing when statements

### PR DESCRIPTION
**Pull request by d4straub**
linting with nf-core tools version 2.5.1 produce a lot of warning, I reduce those here by a large margin:
- added when sections to all modules
- add emit statements
- output versions in all modules

I havent fixed (yet?):

│ assignsh │ modules/local/assignsh.nf│ 'meta' map not emitted in output channel(s)│
│ cutadapt_summary │ modules/local/cutadapt_summary.nf │ 'meta' map not emitted in output channel(s) │
│ qiime2_train │ modules/local/qiime2_train.nf │ 'meta' map not emitted in output channel(s) │
│ sbdiexport │ modules/local/sbdiexport.nf │ Conda version not specified correctly │
│ sbdiexportreannotate │ modules/local/sbdiexportreannotate.nf │ Conda version not specified correctly 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/novaseq_err.nf [hunk 1]
```diff
@@ -13,16 +13,29 @@ process NOVASEQ_ERR {
...
+        END_VERSIONS
         """
     } else {
         """
         novaseq_err_se.r ${errormodel} ${meta.run}
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            R: \$(R --version | sed -n 1p | sed 's/R version //g' | sed 's/\\s.*\$//')
+        END_VERSIONS
```
Comment (erikrikarddaniel) [inline modules/local/novaseq_err.nf:38]: The two version calls are identical, right? If so, one block after the if clause would do it, I guess.

**Review by erikrikarddaniel — APPROVED** [review]
There's a price for standardisation, I suppose ;-)
Just one tiny comment, otherwise LGTM.

File: modules/local/novaseq_err.nf (hunk 1 above)
Comment (d4straub) [reply modules/local/novaseq_err.nf:38]: Yes, the version calls are identical. However, they are in groovy if/else and separate bash blocks, so I am actually not sure whether that works after the if clause. I'll test on Friday!

Comment (erikrikarddaniel) [reply modules/local/novaseq_err.nf:38]: I see. Not worth it I suppose.

**Comment by d4straub** [comment]
Thanks!

**Pull request merged**